shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/compiler/SampleNodeSquish.java

```diff
@@ -31,6 +31,14 @@ public void collectFreeVariables(Set<String> names) {
 
 	}
 
+	@Override
+	public Consumption consumptionCheck(Consumption previous, Consumer<String> errorHandler) {
+		if (previous == Consumption.BAD) {
+			return Consumption.BAD;
+		}
+		return Consumption.VARIABLE;
```

Comment:
so does this mean that it can guarantee that we only allow `Squish` to be called after `Fixed` or `FixedWithCondition`?

Reply:
(and `Squish` can be the first subsampler without being followed by `Fixed` or `FixedWithCondition`.)

Reply:
Yes, that's correct.